pkg/api/norman/customization/aks/handler.go

```diff
@@ -126,6 +140,52 @@ func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (h *handler) checkCredentials(req *http.Request) (int, error) {
+	cred := &Capabilities{}
+	raw, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot read request body: %v", err)
+	}
+
+	if err = json.Unmarshal(raw, &cred); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot parse request body: %v", err)
+	}
+
+	if cred.SubscriptionID == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide subscriptionId")
+	}
+	if cred.ClientID == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide clientId")
+	}
+	if cred.ClientSecret == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide clientSecret")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	if cred.TenantID == "" {
+		cred.TenantID, err = azureutil.FindTenantID(ctx, azure.PublicCloud, cred.SubscriptionID)
+		if err != nil {
+			return http.StatusBadRequest, err
+		}
+	}
+	if cred.BaseURL == "" {
+		cred.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+	if cred.AuthBaseURL == "" {
+		cred.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+	client, err := NewSubscriptionServiceClient(cred)
+	if err != nil {
+		return http.StatusBadRequest, err
```

Comment:
On the errors convo, agree that its not huge concern, but long term I think we should obscure the errors more. And for this specific method we should make errors related to whether the creds are valid or not, so at this point and below just say "Invalid creds". The checks above this point about not passing enough params or invalid body are ok.

Reply:
done